packages/material-ui-styles/src/index.js

```diff
@@ -1,7 +1,7 @@
 import { ponyfillGlobal } from '@material-ui/utils';
 
 /* Warning if there are several instances of @material-ui/styles */
-if (process.env.NODE_ENV !== 'production') {
+if (process.env.NODE_ENV !== 'production' && process.env.NODE_ENV !== 'test') {
```

Comment:
The warning is raised in the watch mode of mocha.

(It will probably not happen in Jest as the global scope is reset between each test.)

Reply:
Probably still some cleanup issues in our tests. I'll push #15387 along and implement automatic cleanup between every tests. We really should isolate each test properly. They shouldn't share state with each other to begin with. With proper isolation we'll have another reason to switch to jest.

Reply:
The pros of using Jest:

- I believe the majority of the community is using Jest. It would force us to have a clean integration of our components with it. For instance, it puts us in a position where we can easily use and fix snapshot tests handling. 
- The tests are sandboxed, I don't think that it's that much an issue for us with Mocha. But it's still peace of mind and potential time-saving.
- The tests are run on multiple CPU cores. It might be faster, but a 42s of test duration, it doesn't really matter. 

Cons:

- The migration cost. That's the big one. Does it worth spending 10 hours on a Mocha to Jest migration vs spending 10h on moving the Slide to the core, implementing the range feature. 
- We would give up on cross-browser testing. Do we need it? I think that most of the value comes from testing that the code has no JavaScript errors, rather than asserting the behavior.
- This issue is still present for Mocha users. We would have spotted the issue later on. It would have probably be reported in a GitHub issue. It's a tradeoff, we can only use one testing framework (🤔), better use the most popular?